php/datasource/class-fieldmanager-datasource-user.php

```diff
@@ -114,146 +135,159 @@ public function get_value( $value ) {
-     */
-    public function get_ajax_action() {
-        if ( !empty( $this->ajax_action ) ) return $this->ajax_action;
-        $unique_key = json_encode( $this->query_args );
-        $unique_key .= $this->display_property;
-        $unique_key .= (string) $this->query_callback;
-        return 'fm_datasource_post' . crc32( $unique_key );
-    }
-
-    /**
-     * Delete reciprocal user metadata prior to saving (presave will re-add).
-     * Reciprocal relationships are not possible if we are not storing by ID.
-     * @param array $values new post values
-     * @param array $current_values existing post values
-     */
-    public function presave_alter_values( Fieldmanager_Field $field, $values, $current_values ) {
-		if ( $field->data_type != 'post' || ! $this->reciprocal || 'ID' != $this->store_property ) {
+	}
+
+	/**
+	 * Get users which match this datasource, optionally filtered by
+	 * a search fragment, e.g. for Autocomplete.
+	 *
+	 * @param string $fragment The search string.
+	 * @return array           post_id => post_title for display or AJAX.
+	 */
+	public function get_items( $fragment = null ) {
+		if ( is_callable( $this->query_callback ) ) {
+			return call_user_func( $this->query_callback, $fragment );
+		}
+
+		$default_args = array();
+		$user_args = array_merge( $default_args, $this->query_args );
+		$ret = array();
+
+		if ( $fragment ) {
+			$user_args['search'] = '*' . $fragment . '*';
+		}
+
+		$users = get_users( $user_args );
+		foreach ( $users as $u ) {
+			$ret[ $u->{$this->store_property} ] = $u->{$this->display_property};
+		}
+
+		return $ret;
+	}
+
+	/**
+	 * Get an action to register by hashing (non cryptographically for speed)
+	 * the options that make this datasource unique.
+	 *
+	 * @return string AJAX action.
+	 */
+	public function get_ajax_action() {
+		if ( ! empty( $this->ajax_action ) ) {
+			return $this->ajax_action;
+		}
+		$unique_key = wp_json_encode( $this->query_args );
+		$unique_key .= $this->display_property;
+		$unique_key .= (string) $this->query_callback;
+		return 'fm_datasource_post' . crc32( $unique_key );
+	}
+
+	/**
+	 * Delete reciprocal user metadata prior to saving (presave will re-add).
+	 * Reciprocal relationships are not possible if we are not storing by ID.
+	 *
+	 * @param Fieldmanager_Field $field          The current field.
+	 * @param array              $values         The new values.
+	 * @param array              $current_values The existing values.
+	 * @return string                            Sanitized values.
+	 */
+	public function presave_alter_values( Fieldmanager_Field $field, $values, $current_values ) {
+		if ( 'post' != $field->data_type || ! $this->reciprocal || 'ID' != $this->store_property ) {
 			return $values;
 		}
 
 		if ( ! empty( $current_values ) ) {
 			foreach ( $current_values as $user_id ) {
-				delete_user_meta( $user_id, $this->reciprocal, $field->data_id );
+				delete_metadata( 'user', $user_id, $this->reciprocal, $field->data_id );
```

Comment:
that fools the linter, but doesn't fix the problem. we'll need to do something [like we did in the user context](https://github.com/alleyinteractive/wordpress-fieldmanager/blob/master/php/context/class-fieldmanager-context-user.php#L126-L148)